Entity/CustomItem.php

```diff
@@ -169,6 +174,11 @@ class CustomItem extends FormEntity implements UniqueEntityInterface
      */
     private $customItemLowerReferences;
 
+    private ?string $uniqueHash = null;
+
+    private bool $wasInserted = false;
+    private bool $wasUpdated = false;
```

Comment:
Can these properties be moved to `Entity/CustomItem/CustomItemUpsertTrait.php` too?